rasterio/env.py

```diff
@@ -1,8 +1,11 @@
 """Rasterio's GDAL/AWS environment"""
 
-from functools import wraps
+from functools import wraps, total_ordering
 import logging
 import threading
+import re
```

Comment:
I normally like to have imports at the top, but it occurs to me that this may be heavier than desired here given that `env` is used throughout rasterio.  Would it be appropriate to inline this import within `GDALVersion` instead?

Reply:
@brendan-ward I'm not concerned about importing re here.

Reply:
Is `re` known to be expensive?  I did a bit of loose local testing and it seems to not really matter.